src/compiler/program.ts

```diff
@@ -921,6 +922,10 @@ namespace ts {
             return sourceFilesFoundSearchingNodeModules.get(file.path);
         }
 
+        function isSourceFileDefaultLibrary(file: SourceFile): boolean {
+            return file.fileName === host.getDefaultLibFileName(options);

```

Comment:
Perhaps this should be named `isSourceFileReadonly` or `isSourceFileLockedForEdits` so that we can later use the same functionality to prevent renames in declarations pulled in from node_modules or other `"typeRoots"` locations.